dockerfile/Dockerfile

```diff
@@ -1,136 +1,74 @@
-  apt-get -y install openssh-server && \
-  apt-get -y install python-pip && \
-  pip install kazoo
-
-#8,add dolphinscheduler source code to /opt/dolphinscheduler_source
-ADD . /opt/dolphinscheduler_source
-
-
-#9,backend compilation
-RUN cd /opt/dolphinscheduler_source && \
-    mvn clean package -Prelease -Dmaven.test.skip=true
-
-#10,frontend compilation
-RUN chmod -R 777 /opt/dolphinscheduler_source/dolphinscheduler-ui && \
-    cd /opt/dolphinscheduler_source/dolphinscheduler-ui && \
-    rm -rf /opt/dolphinscheduler_source/dolphinscheduler-ui/node_modules && \
-    npm install node-sass --unsafe-perm && \
-    npm install && \
-    npm run build
-
-#11,modify dolphinscheduler configuration file
-#backend configuration
-RUN tar -zxvf /opt/dolphinscheduler_source/dolphinscheduler-dist/dolphinscheduler-backend/target/apache-dolphinscheduler-incubating-${tar_version}-dolphinscheduler-backend-bin.tar.gz -C /opt && \
-    mv /opt/apache-dolphinscheduler-incubating-${tar_version}-dolphinscheduler-backend-bin /opt/dolphinscheduler && \
-    rm -rf /opt/dolphinscheduler/conf
-
-ADD ./dockerfile/conf/dolphinscheduler/conf /opt/dolphinscheduler/conf
-#frontend nginx configuration
-ADD ./dockerfile/conf/nginx/dolphinscheduler.conf /etc/nginx/conf.d
-
-#12,open port
-EXPOSE 2181 2888 3888 3306 80 12345 8888
-
-COPY ./dockerfile/startup.sh /root/startup.sh
-#13,modify permissions and set soft links
-RUN chmod +x /root/startup.sh && \
-  chmod +x /opt/dolphinscheduler/script/create-dolphinscheduler.sh && \
-  chmod +x /opt/zookeeper/bin/zkServer.sh && \
-  chmod +x /opt/dolphinscheduler/bin/dolphinscheduler-daemon.sh && \
-  rm -rf /bin/sh && \
-  ln -s /bin/bash /bin/sh && \
-  mkdir -p /tmp/xls
-
+ADD ./conf/zookeeper/zoo.cfg /opt/zookeeper/conf
+ENV ZK_HOME /opt/zookeeper
+ENV PATH $ZK_HOME/bin:$PATH
+
+#4. install pg
+RUN apk add postgresql postgresql-contrib
+
+#5. add dolphinscheduler
+ADD ./apache-dolphinscheduler-incubating-${VERSION}-SNAPSHOT-dolphinscheduler-bin.tar.gz /opt/
+RUN mv /opt/apache-dolphinscheduler-incubating-${VERSION}-SNAPSHOT-dolphinscheduler-bin/ /opt/dolphinscheduler/
+ENV DOLPHINSCHEDULER_HOME /opt/dolphinscheduler
+
+#6. modify nginx
+RUN echo "daemon off;" >> /etc/nginx/nginx.conf && \
+    rm -rf /etc/nginx/conf.d/*
+ADD ./conf/nginx/dolphinscheduler.conf /etc/nginx/conf.d
+
+#7. add configuration and modify permissions and set soft links
+ADD ./startup-init-conf.sh /root/startup-init-conf.sh
+ADD ./startup.sh /root/startup.sh
+ADD ./conf/dolphinscheduler/*.tpl /opt/dolphinscheduler/conf/
+ADD ./conf/dolphinscheduler/env/dolphinscheduler_env /opt/dolphinscheduler/conf/env/
+RUN chmod +x /root/startup-init-conf.sh && \
+    chmod +x /root/startup.sh && \
+    chmod +x /opt/dolphinscheduler/conf/env/dolphinscheduler_env && \
+    chmod +x /opt/dolphinscheduler/script/*.sh && \
+    chmod +x /opt/dolphinscheduler/bin/*.sh && \
+    chmod +x /opt/zookeeper/bin/*.sh && \
+    dos2unix /root/startup-init-conf.sh && \
```

Comment:
Why dos2unix, not sh?

Reply:
Because bash script with Windows line-ending `CRLF` won't work on Linux.

UNIX and UNIX-like operating systems (including Mac OS X) represent line endings as `LF` alone.

Converting line-ending using `dos2unix`.